CI change for mergifyio `strict` mode deprecation

humblec opened the pull request:
ci: move from merge action to queue action
    
    as mentioned in the below blog the support for strict mode
    and merge action will be done soon in mergify. This brings
    the change requested for the same.
    
    Ref# https://blog.mergify.com/strict-mode-deprecation/

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

nixpanic reviewed (changes requested):
Have you tested this in an other repository already? The multi-queue configuration requires to have the premium feature plan... We need to setup a single queue instead.

https://docs.mergify.com/actions/queue/#examples

humblec commented:
> Have you tested this in an other repository already? The multi-queue configuration requires to have the premium feature plan... We need to setup a single queue instead.
> 
> https://docs.mergify.com/actions/queue/#examples

@nixpanic thanks.. missed this `plan` thing.. will revisit this PR for the single queue option then. looks difficult as the conditions are different for each branch. but will give a try

Madhu-1 commented:
>The configuration uses the deprecated strict mode of the merge action.
A brownout is planned for the whole December 6th, 2021 day.
This option will be removed on January 10th, 2022.
For more information: https://blog.mergify.com/strict-mode-deprecation/

@humblec this is needed as mergify configuration is not considered from today, can you please check on this one?

humblec commented:
Yeah, today is the deprecation. looks like we are in that. eventhough its still 10th :)

humblec committed:
ci: move from merge action to queue action

as mentioned in the below blog the support for strict mode
and merge action will be done soon in mergify. This brings
the change requested for the same.

Ref# https://blog.mergify.com/strict-mode-deprecation/

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
hmm.. the error is confusing in `summary` check:

It is giving:
```


    The configuration uses the deprecated strict mode of the merge action.
    A brownout is planned for the whole December 6th, 2021 day.
    This option will be removed on January 10th, 2022.
    For more information: https://blog.mergify.com/strict-mode-deprecation/

rebase is invalid for dictionary value @ defaults → actions → merge → strict_method

    The configuration uses the deprecated strict mode of the merge action.
    A brownout is planned for the whole December 6th, 2021 day.
    This option will be removed on January 10th, 2022.
    For more information: https://blog.mergify.com/strict-mode-deprecation/

smart is invalid for dictionary value @ defaults → actions → merge → strict

```

The `strict` or `strict_method` is not part of the rules any more.

Can I have second pair of eyes on this @ceph/ceph-csi-maintainers

Madhu-1 commented:
https://dashboard.mergify.com/github/ceph/repo/ceph-csi/config-editor configurations look valid to me.

Rakshith-R commented on .mergify.yml:
```diff
@@ -4,22 +4,51 @@ defaults:
+      update_method: rebase
+queue_rules:
+  - name: default
+    conditions:
+      # Conditions to get out of the queue (= merged)
+      - label!=DNM
+      - base~=^(devel)|(release-.+)$
+      - "#approved-reviews-by>=2"
+      - "#changes-requested-reviews-by=0"
+      - "approved-reviews-by=@ceph/ceph-csi-contributors"
+      - "approved-reviews-by=@ceph/ceph-csi-maintainers"
+      - "status-success=codespell"
+      - "status-success=multi-arch-build"
+      - "status-success=go-test"
+      - "status-success=golangci-lint"
+      - "status-success=commitlint"
+      - "status-success=mod-check"
+      - "status-success=lint-extras"
+      - "status-success=ci/centos/k8s-e2e-external-storage/1.21"
+      - "status-success=ci/centos/k8s-e2e-external-storage/1.22"
+      - "status-success=ci/centos/mini-e2e-helm/k8s-1.20"
+      - "status-success=ci/centos/mini-e2e-helm/k8s-1.21"
+      - "status-success=ci/centos/mini-e2e-helm/k8s-1.22"
+      - "status-success=ci/centos/mini-e2e-helm/k8s-1.23"
+      - "status-success=ci/centos/mini-e2e/k8s-1.20"
+      - "status-success=ci/centos/mini-e2e/k8s-1.21"
+      - "status-success=ci/centos/mini-e2e/k8s-1.22"
+      - "status-success=ci/centos/mini-e2e/k8s-1.23"
+      - "status-success=ci/centos/upgrade-tests-cephfs"
+      - "status-success=ci/centos/upgrade-tests-rbd"
+      - "status-success=DCO"
```
```suggestion
      - or:
        - and:
      - "status-success=codespell"
      - "status-success=multi-arch-build"
      - "status-success=go-test"
      - "status-success=golangci-lint"
      - "status-success=commitlint"
      - "status-success=mod-check"
      - "status-success=lint-extras"
      - "status-success=ci/centos/k8s-e2e-external-storage/1.21"
      - "status-success=ci/centos/k8s-e2e-external-storage/1.22"
      - "status-success=ci/centos/mini-e2e-helm/k8s-1.20"
      - "status-success=ci/centos/mini-e2e-helm/k8s-1.21"
      - "status-success=ci/centos/mini-e2e-helm/k8s-1.22"
      - "status-success=ci/centos/mini-e2e-helm/k8s-1.23"
      - "status-success=ci/centos/mini-e2e/k8s-1.20"
      - "status-success=ci/centos/mini-e2e/k8s-1.21"
      - "status-success=ci/centos/mini-e2e/k8s-1.22"
      - "status-success=ci/centos/mini-e2e/k8s-1.23"
      - "status-success=ci/centos/upgrade-tests-cephfs"
      - "status-success=ci/centos/upgrade-tests-rbd"
      - "status-success=DCO"
```
Will not work for ci/centos branch, 
I'll push change into this pr in a min

humblec replied:
ah.. yes, ci branch need changes. @Rakshith-R please feel free to push the changes :+1:

Madhu-1 commented on .mergify.yml:
```diff
@@ -4,22 +4,51 @@ defaults:
-    # bot_account: ceph-csi-bot # same as update_bot_account.
-
+      update_method: rebase
+queue_rules:
```
this might not work for the release-3.4 branch as we don't run k8s-e2e-external-storage tests on it and this won't work for the centos/ci branch as we don't run any of these tests on it.

humblec replied:
as @Rakshith-R  mentioned, centos ci branch need changes. will wait for the PR refresh, it should be covered in it.

humblec commented:
> https://dashboard.mergify.com/github/ceph/repo/ceph-csi/config-editor configurations look valid to me.

yeah, but the summary test gives `invalid` . was confused.

humblec commented:
By the proposed changes of conditions, this looks good now.. ptal thanks..

humblec commented:
@Madhu-1 ptal ..

Madhu-1 commented on .mergify.yml:
```diff
@@ -4,22 +4,73 @@ defaults:
+              - "status-success=ci/centos/upgrade-tests-cephfs"
+              - "status-success=ci/centos/upgrade-tests-rbd"
+              - "status-success=DCO"
+              - "status-success=DCO"
```
duplicate check

Madhu-1 commented on .mergify.yml:
```diff
@@ -4,22 +4,73 @@ defaults:
+              - "status-success=golangci-lint"
+              - "status-success=mod-check"
+              - "status-success=lint-extras"
+              - "#changes-requested-reviews-by=0"
```
this is not required

humblec replied:
removed

Madhu-1 commented on .mergify.yml:
```diff
@@ -4,22 +4,75 @@ defaults:
+              - "status-success=DCO"
+          - and:
+              - base=ci/centos
+              - "approved-reviews-by=@ceph/ceph-csi-maintainers"
+              - "#changes-requested-reviews-by=0"
```
this check is not required.

humblec replied:
Done

Madhu-1 reviewed (changes requested):
@humblec are these mergify changes validated with some test repository?

humblec commented:
@Madhu-1 ptal.. thanks.

humblec commented:
> @humblec are these mergify changes validated with some test repository?

no @Madhu-1

Madhu-1 commented on .mergify.yml:
```diff
@@ -4,22 +4,72 @@ defaults:
+              - "status-success=DCO"
+          - and:
+              - base=ci/centos
+              - "approved-reviews-by=@ceph/ceph-csi-maintainers"
```
looks like this is not removed yet.

humblec replied:
should be good now @Madhu-1
